### ISSUE TITLE
Previously completed transcriptions disappear when refreshing Whishper homepage.

### ISSUE THREAD

**Issue by JohnDoe-dev-spec** [pluja/whishper#14]
When i transcribe a file and refresh the home page it disappears for some unknown reason and i can no longer see,edit or download it.

Whishper-Web Logs
```
2023-08-30T22:00:46.987141000Z Listening on 0.0.0.0:3000
2023-08-30T22:01:36.731874000Z /api/transcriptions
2023-08-30T22:01:36.850155000Z TypeError: fetch failed
2023-08-30T22:01:36.850330000Z     at fetch (file:///app/shims.js:20346:13)
2023-08-30T22:01:36.850404000Z     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2023-08-30T22:01:36.850484000Z     at async resolve (file:///app/server/index.js:3998:14)
2023-08-30T22:01:36.850573000Z     at async respond (file:///app/server/index.js:3823:22)
2023-08-30T22:01:36.850651000Z     at async fetch (file:///app/server/index.js:3633:26)
2023-08-30T22:01:36.850714000Z     at async Object.fetch (file:///app/server/index.js:3573:12)
2023-08-30T22:01:36.850777000Z     at async file:///app/server/index.js:1996:20
2023-08-30T22:01:36.850840000Z     at async load (file:///app/server/chunks/0-9ff9db82.js:9:20)
2023-08-30T22:01:36.850903000Z     at async load_data (file:///app/server/index.js:1978:18)
2023-08-30T22:01:36.850955000Z     at async file:///app/server/index.js:3307:18 {
2023-08-30T22:01:36.851022000Z   cause: [Error: 989BD3810F7F0000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:354:
2023-08-30T22:01:36.851097000Z   ] {
2023-08-30T22:01:36.851159000Z     library: 'SSL routines',
2023-08-30T22:01:36.851225000Z     reason: 'wrong version number',
2023-08-30T22:01:36.851286000Z     code: 'ERR_SSL_WRONG_VERSION_NUMBER'
2023-08-30T22:01:36.851360000Z   }
2023-08-30T22:01:36.851429000Z 
```

Whishper-backend Logs

```
2023-08-30T22:00:46.836496000Z 22:00:46 DBG main.go:53 > DEV MODE IS ON
2023-08-30T22:00:46.836672000Z 22:00:46 DBG main.go:54 > ListenAddr: :8080
2023-08-30T22:00:46.836743000Z 22:00:46 DBG main.go:55 > UploadDir: /app/uploads
2023-08-30T22:00:46.836792000Z 22:00:46 DBG main.go:56 > AsrEndpoint: transcription-api:8000
2023-08-30T22:00:46.836850000Z 22:00:46 DBG main.go:57 > TranslationEndpoint: translate:5000
2023-08-30T22:00:46.836920000Z 22:00:46 DBG main.go:58 > DbHost: mongo:27017
2023-08-30T22:00:46.837012000Z 
2023-08-30T22:00:46.837209000Z  ┌───────────────────────────────────────────────────┐ 
2023-08-30T22:00:46.837352000Z  │                   Fiber v2.48.0                   │ 
2023-08-30T22:00:46.837514000Z  │               http://127.0.0.1:8080               │ 
2023-08-30T22:00:46.837675000Z  │       (bound on host 0.0.0.0 and port 8080)       │ 
2023-08-30T22:00:46.837828000Z  │                                                   │ 
2023-08-30T22:00:46.837984000Z  │ Handlers ............ 13  Processes ........... 1 │ 
2023-08-30T22:00:46.838141000Z  │ Prefork ....... Disabled  PID ................. 1 │ 
2023-08-30T22:00:46.838328000Z  └───────────────────────────────────────────────────┘ 
2023-08-30T22:00:46.838460000Z 
2023-08-30T22:00:46.838550000Z 22:00:46 INF monitor/monitor.go:18 > Starting monitor!
2023-08-30T22:00:46.866392000Z 22:00:46 DBG monitor/monitor.go:25 > Pending transcriptions: 0
2023-08-30T22:01:42.943467000Z 22:01:42 DBG api/server.go:122 > POST /api/transcriptions
2023-08-30T22:01:42.943865000Z 22:01:42 DBG api/handlers.go:56 > POST /api/transcriptions
2023-08-30T22:01:42.943956000Z 22:01:42 DBG api/handlers.go:93 > Transcription: {ID:ObjectID("000000000000000000000000") Status:0 Language:en ModelSize:small Task:transcribe Device:cpu FileName: SourceUrl:https://www.youtube.com/watch?v=WOb5GRQ0VFE Result:{Language: Duration:0 Segments:[] Text:} Translations:[]}
2023-08-30T22:01:42.946713000Z 22:01:42 DBG monitor/monitor.go:25 > Pending transcriptions: 1
2023-08-30T22:01:42.946827000Z 22:01:42 DBG monitor/monitor.go:27 > Taking pending transcription ObjectID("64efbc463a2bf39da6533ad4")
2023-08-30T22:01:42.946893000Z 22:01:42 DBG monitor/monitor.go:49 > Updating transcription &{ObjectID("64efbc463a2bf39da6533ad4") 1 en small transcribe cpu  https://www.youtube.com/watch?v=WOb5GRQ0VFE { 0 [] } []}
2023-08-30T22:02:52.120757000Z 22:02:52 ERR api/server.go:58 > Error reading message error="websocket: close 1006 (abnormal closure): unexpected EOF"
2023-08-30T22:04:39.799407000Z 22:04:39 ERR api/server.go:58 > Error reading message error="websocket: close 1006 (abnormal closure): unexpected EOF"
```

Thanks!

**Comment by pluja** [comment]
How are you hosting whishper? Did you correctly set the `WHISHPER_HOST` variable in the `.env`?

**Comment by JohnDoe-dev-spec** [comment]
> How are you hosting whishper? Did you correctly set the `WHISHPER_HOST` variable in the `.env`?

Hi Pluja,

I am running Whishper on DietPi Debian OS.

The docker-compose.yml is default with no changes either.

The `WHISPER_HOST` is default, `WHISHPER_HOST=localhost:8082`

I'm accessing Whishper via http://IPADDRESS:8082

Thanks

**Comment by pluja** [comment]
In this case, can you try to set WHISHPER_HOST to `IPADDRESS:8082` instead of `localhost:8082` since you're accessing from your IP address?

**Comment by JohnDoe-dev-spec** [comment]
Just tried now and i'm afraid the same issue. After successful transcribe, i refresh the page and it disappears.

**Comment by pluja** [comment]
Can you share the size of the file you're trying to transcribe? I think there might be issues with large files

**Comment by JohnDoe-dev-spec** [comment]
It's fairly small, first one is a link to a three minute YouTube video.

The other one is maybe 15MB Audio file.

**Comment by pluja** [comment]
Can you check if `http://IPADDRESS:8082/editor/64efbc463a2bf39da6533ad4` (I got the transcription id from your logs) shows the transcription?

**Comment by JohnDoe-dev-spec** [comment]
I get a 500 Internal Error page.

I just tried running the same transcribe again, screenshots 1 and 2 are before i do a refresh on the page. Then i refresh and try the transcription Id and get above error.

[](https://ibb.co/bJvVJg6
https://ibb.co/wCKzs5m
https://ibb.co/sthk26S)

**Comment by pluja** [comment]
That's weird, I will investigate and replicate this tomorrow and try to push a fix as soon as possible.

**Comment by eyalspier** [comment]
+1 for the exact same issue on my end!

**Comment by Fabiettus** [comment]
Same behavior here!

**Comment by North-West-Wind** [comment]
If you manage to get back into the editor and save your transcription, it will re-appear on the home page.

> I get a 500 Internal Error page.

If you have kept the tab opened in your browser, you can try using history forward and backward and you can get back into the editor.

**Comment by pluja** [comment]
Hi, all, I will have to investigate this further, can you tell me how are you hosting whishper? Do you see any errors on the docker logs? And in the browser console logs are there any errors? I will need more information since I am not being able to replicate this issue in my local environment.

**Comment by eyalspier** [comment]
I've also spent some time digging around:
When I add a WHISPER_PUBLIC_URL variable, I get a 1006 error. Attempting to access http://mydomain/ws/transcriptions gives me an "upgrade required" message.

**Comment by Pant** [comment]
# TLDR; 

## 1. Run the following command in `frontend` directory:

```
docker build -t pluja/whishper-frontend:latest  --build-arg PUBLIC_INTERNAL_API_HOST=http://whishper-backend:8080 .  
```

## 2. Stop and Start docker compose:

```
docker compose stop
docker compose up -d
```

---

# Issue
##  PUBLIC_INTERNAL_API_HOST has no value

In the logs of the first comment in this issue, the log message of interest is: `/api/transcriptions`. Here there is no host in the URL.

Eventually i figured that this log message comes from `console.log(endpoint);` in file `routes/+layout.js`.

## Change in `docker-compose.yml` env variable.

Changing the value in the `docker-compose.yml` is not changing anything.

## Solution

Add the argument in build time, as describe in TLDR section.

**Comment by pluja** [comment]
@Pant  thank you for pointing out the problem and the fix.  I just pushed an update that should fix all of this. Make sure to update the docker-compose.yml `whishper-frontend` variable names.

**Comment by JohnDoe-dev-spec** [comment]
All resolved here too now with latest compose.

Thanks both for resolving!

**Comment by eyalspier** [comment]
Also confirm it works now - thank you so much!
